src/p_pspr.c

```diff
@@ -80,7 +80,8 @@ static void P_SetPsprite(player_t *player, int position, statenum_t stnum)
 
   if (position == ps_weapon)
   {
-    if (stnum == weaponinfo[player->pendingweapon].upstate)
+    if (stnum == weaponinfo[player->pendingweapon].upstate ||
```

Comment:
For some reason I am stubborn to think that `pendingweapon` needs to be checked as well.

Reply:
Thanks for the review! Wow, you are right that `player->pendingweapon == wp_nochange` would overflow the `weaponinfo[]` array. Good catch! 

However, I somehow had in the depths of my mind that you start the game with a pending weapon and don't have a ready weapon yet. But maybe I'm just wrong. 🤔

Reply:
> However, I somehow had in the depths of my mind that you start the game with a pending weapon and don't have a ready weapon yet. But maybe I'm just wrong. thinking

I did a bit of digging, and apparently, the following happens when starting a new game:

- `G_InitNew()` sets `player->playerstate = PST_REBORN` for all players
- Due to the former, `P_SpawnPlayer()` calls `G_PlayerReborn()` for said players
- The latter sets `p->readyweapon = p->pendingweapon = wp_pistol`
- Later on in `P_SpawnPlayer()`, `P_SetupPsprites()` is called
- The latter sets `player->pendingweapon = player->readyweapon` and calls `P_BringUpWeapon()`

TL;DR, you start the new game with the Pistol as both your `readyweapon` and `pendingweapon`. I'm assuming that `readyweapon` is just carried over when changing levels.

Reply:
So, you start with a `pendingweapon`, but it's the same as `readyweapon`. So the check for the former is at least redundant, if not even dangerous if it is set to `wp_nochange`. Thanks for clarifying!